packages/arb-token-bridge-ui/src/hooks/useChainId.ts

```diff
@@ -5,11 +5,20 @@ export function useChainId({ provider }: { provider: providers.Provider }) {
   const [chainId, setChainId] = useState<number | undefined>(undefined)
 
   useEffect(() => {
+    let cancelled = false
```

Comment:
Fixes an issue where sometimes `useBalance` would return an outdated result. Here we make sure we always get the latest.

It wasn't much of an issue before, but with Orbit chains, Arbitrum can flip from being L2 to L1, or L1 to L2, nearly instantaneously, which makes this bug more prevalent here.